Makefile

```diff
@@ -78,22 +78,22 @@ all-srcs:
 .PHONY: build
 build: install-tools lint multimod-verify
 	GOOS=darwin GOARCH=amd64 $(GOBUILD) $(LDFLAGS) -o ./build/darwin/aoc_darwin_amd64 ./cmd/awscollector
-	GOOS=linux GOARCH=amd64 $(GOBUILD) $(LDFLAGS) -o ./build/linux/aoc_linux_x86_64 ./cmd/awscollector
```

Comment:
Have we talked with @alolita  @anuraaga @Aneurysm9 that being backwards not compatible is okay here

Reply:
Is this just an intermediate name used during the build process?  Does the name of the executable included in the `deb`, `rpm`, or container image change?

Reply:
Yes. This is just an intermediate name. `create_rpm.sh` and `create_deb.sh` both copy the binary in and rename it to `aws-otel-collector`. The Dockerfile also renames it, but to `awscollector`.

Reply:
Yeah intermediate names should not need to be preserved.

While we're changing these, can we dedupe, and maybe simplify, by moving into the folder

`./build/linux-amd64/aoc`?